frontend/src/app/my/mentorship/programs/[programKey]/modules/[moduleKey]/issues/[issueId]/page.tsx

```diff
@@ -0,0 +1,501 @@
+        color: 'success',
+        timeout: 2500,
+        shouldShowTimeoutProgress: true,
+      })
+      setIsEditingDeadline(false)
+    },
+    onError: (err) => {
+      addToast({
+        title: 'Failed to update deadline: ' + err.message,
+        variant: 'solid',
+        color: 'danger',
+        timeout: 3500,
+        shouldShowTimeoutProgress: true,
+      })
+    },
+  })
+
+  if (error) {
+    return <ErrorDisplay statusCode={500} title="Error Loading Issue" message={error.message} />
+  }
+  if (loading) return <LoadingSpinner />
+  if (!issue)
+    return <ErrorDisplay statusCode={404} title="Issue Not Found" message="Issue not found" />
+
+  const assignees = issue.assignees || []
+  const labels = issue.labels || []
+  const visibleLabels = labels.slice(0, 5)
+  const remainingLabels = labels.length - visibleLabels.length
+  const canEditDeadline = assignees.length > 0
+
+  return (
+    <div className="min-h-screen bg-white p-8 text-gray-700 dark:bg-[#212529] dark:text-gray-300">
+      <div className="mx-auto max-w-5xl">
+        <div className="mb-6 flex flex-col gap-4 sm:flex-row sm:items-center sm:justify-between">
+          <div className="min-w-0 flex-1">
+            <h1 className="text-2xl font-bold sm:text-3xl">
+              <span className="break-words">{issue.title}</span>
+            </h1>
+            <div className="mt-1 flex items-center gap-2 text-sm text-gray-500">
+              <span>
+                {issue.organizationName}/{issue.repositoryName} • #{issue.number}
+              </span>
+              <span
+                className={`inline-flex items-center rounded-lg px-2 py-0.5 text-xs font-medium ${
+                  issue.state === 'open' ? 'bg-[#2cbe4e] text-white' : 'bg-[#cb2431] text-white'
+                }`}
+              >
+                {issue.state === 'open' ? 'Open' : 'Closed'}
+              </span>
+            </div>
+          </div>
+          <ActionButton url={issue.url} tooltipLabel="View on GitHub">
+            <FontAwesomeIcon icon={faLink} /> View on GitHub
+          </ActionButton>
+        </div>
+
+        <SecondaryCard title={<AnchorTitle title="Description" />}>
+          <div className="prose dark:prose-invert line-clamp-[15] max-w-none">
+            <Markdown content={issue.body || 'No description.'} />
+          </div>
+        </SecondaryCard>
+
+        <SecondaryCard title={<AnchorTitle title="Task Timeline" />}>
+          <div className="space-y-4 text-sm text-gray-700 dark:text-gray-300">
+            <div>
+              <span className="font-medium">Assigned:</span>{' '}
+              {data?.getModule?.taskAssignedAt
+                ? new Date(data.getModule.taskAssignedAt).toLocaleDateString()
+                : 'Not assigned'}
+            </div>
+
+            <div className="flex flex-col gap-2">
+              <div className="flex flex-wrap items-center gap-2">
+                <span className="font-medium">Deadline:</span>
```

Comment:
Can we simplify this logic so that the edit form opens up as soon as the user clicks the "No deadline set" or Current deadline date? Meaning that whole thing would be like a button trigger.
We won't need `edit` button this way. 

We also won't need `close` button for editing. To cancel the change user can click out of the "calendar" view, use `Clear` on it or edit manually. 
And we will accept the edit as soon as user selects the date. This should set it to deadline automatically, without having to click `Save`.

This will simplify the view a lot!